### DISCUSSION TITLE
How does CEA create PV energy supply?

### DISCUSSION THREAD

**Discussion by KQEFEM** [architecture-building-systems/CityEnergyAnalyst#3317] [Q&A] [answered]
Hi All, 
I have a question about how the CEA uses PV:

1. Does it just assume that every building can have x% PV panels on the roofs
2. Is there a way of saying building 1 has x% and building 2 has y%? 

I am using the Dashboard for my simulations, I can use a Python IDE if that is easier.

**Comment by lguilhermers** [comment, marked as answer]
Hi Keiran, I think the PV generator was developed as a more general tool so it does not cater so easily for per-building customization.
While in theory it could work like that (and in my point of view it would be even better), it would need code extension.
If you are familiar with python you can try to create a branch that creates that, but there would be some complexity involved.

A quick alternative, which might not be as accurate, is to model all buildings with 100% and, in post-processing you multiply by a factor that is equivalent to the share of PVs you want to apply for each building.
